tests/mongodb/test_mongodb_source.py

```diff
@@ -0,0 +1,30 @@
+from tests.utils import ALL_DESTINATIONS, assert_load_info, load_table_counts
+import pytest
+import dlt
+from sources.mongodb import mongodb
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_all_resources(destination_name: str) -> None:
+    pipeline = dlt.pipeline(
+        pipeline_name="mongodb",
+        destination=destination_name,
+        dataset_name="mongodb_data",
+        full_refresh=True,
+    )
+    load_info = pipeline.run(mongodb())
+    print(load_info)
+    assert_load_info(load_info)
+    table_names = [t["name"] for t in pipeline.default_schema.data_tables()]
+    table_counts = load_table_counts(pipeline, *table_names)
+
+    # for now only check main tables
+    expected_tables = ["collection1", "collection2"]
+    assert set(table_counts.keys()) >= set(expected_tables)
+
+    assert table_counts["collection1"] == 20
```

Comment:
how does this test work? do we need to use mongo mock to create collections being tested? I do not see that here

Reply:
when we create mock collections let's make sure all data types (including dates, decimals, nested jsons are there)